src/libs/elektra/kdb.c

```diff
@@ -362,591 +363,790 @@ static int ensureContract (KDB * handle, const KeySet * contract, Key * parentKe
-				/* TODO: Remove usage of deprecated internal iterator */
-				ksRewind (ks);
-				handle->globalPlugins[PROCGETSTORAGE][FOREACH]->kdbGet (handle->globalPlugins[PROCGETSTORAGE][FOREACH], ks,
-											parentKey);
-				keySetName (parentKey, keyName (split->parents[i]));
-			}
-			if (p == (STORAGE_PLUGIN + 2) && handle->globalPlugins[POSTGETSTORAGE][FOREACH])
-			{
-				keySetName (parentKey, keyName (initialParent));
-				/* TODO: Remove usage of deprecated internal iterator */
-				ksRewind (ks);
-				handle->globalPlugins[POSTGETSTORAGE][FOREACH]->kdbGet (handle->globalPlugins[POSTGETSTORAGE][FOREACH], ks,
-											parentKey);
-				keySetName (parentKey, keyName (split->parents[i]));
-			}
-			else if (p == (NR_OF_PLUGINS - 1) && handle->globalPlugins[POSTGETCLEANUP][FOREACH])
-			{
-				keySetName (parentKey, keyName (initialParent));
-				/* TODO: Remove usage of deprecated internal iterator */
-				ksRewind (ks);
-				handle->globalPlugins[POSTGETCLEANUP][FOREACH]->kdbGet (handle->globalPlugins[POSTGETCLEANUP][FOREACH], ks,
-											parentKey);
-				keySetName (parentKey, keyName (split->parents[i]));
-			}
+	int errnosave = errno;
 
-			if (backend->getplugins[p] && backend->getplugins[p]->kdbGet)
-			{
-				if (p <= STORAGE_PLUGIN)
-				{
-					if (!test_bit (split->syncbits[i], SPLIT_FLAG_SYNC))
-					{
-						// skip it, update is not needed
-						continue;
-					}
-
-					ret = backend->getplugins[p]->kdbGet (backend->getplugins[p], split->keysets[i], parentKey);
-				}
-				else
-				{
-					KeySet * cutKS = prepareGlobalKS (ks, parentKey);
-					ret = backend->getplugins[p]->kdbGet (backend->getplugins[p], cutKS, parentKey);
-					ksAppend (ks, cutKS);
-					ksDel (cutKS);
-				}
-			}
+	// Step 1: create empty KDB instance
+	KDB * handle = kdbNew (errorKey);
+	if (handle == NULL)
+	{
+		goto error;
+	}
 
-			if (ret == -1)
-			{
-				keySetName (parentKey, keyName (initialParent));
-				// Ohh, an error occurred,
-				// lets stop the process.
-				elektraGlobalError (handle, ks, parentKey, GETSTORAGE, DEINIT);
-				// elektraGlobalError (handle, ks, parentKey, POSTGETSTORAGE, DEINIT);
-				return -1;
-			}
-		}
+	// Step 2: configure for bootstrap
+	if (!addElektraMountpoint (handle->backends, handle->modules, handle->global, errorKey))
+	{
+		goto error;
 	}
 
-	if (run == FIRST)
+	// Step 3: execute bootstrap
+	KeySet * elektraKs = elektraBoostrap (handle, errorKey);
+	if (elektraKs == NULL)
 	{
-		keySetName (parentKey, keyName (initialParent));
-		elektraGlobalGet (handle, ks, parentKey, GETSTORAGE, DEINIT);
-		// elektraGlobalGet (handle, ks, parentKey, POSTGETSTORAGE, DEINIT);
+		goto error;
 	}
-	return 0;
-}
 
-static int copyError (Key * dest, Key * src)
-{
-	const Key * metaKey = keyGetMeta (src, "error");
-	if (!metaKey) return 0;
-	keySetMeta (dest, keyName (metaKey), keyString (metaKey));
+	// Step 4: process contract and set up hooks
+	// TODO [new_backend]: remove once all globals are replacd by hooks
```

Comment:
Is there still some working global plugin?

```suggestion
	// TODO [new_backend]: remove once all globals are replaced by hooks
```

Reply:
No not really. And even if, most wouldn't work as we do not call their `get` or `set` functions. The only global plugins that would be affected by this are the notification receive plugins, as they only operate on the plugin `open` function.

Reply:
Okay, I will remove all the global plugin code. If it breaks any additional tests, I will just disable them.

Reply:
Best would be if you also remove the file `global.c`.

Reply:
Maybe @atmaxinger can help out here? (by simply pushing to the same branch?)

Reply:
I think only 1 person should work on this. But I'd gladly do it if @kodebach hasn't started yet.

Reply:
@atmaxinger go ahead